### PR TITLE
Migrate TEFCA Viewer to Next JS

### LINKED ISSUE #1730: Merge current TEFCA Viewer progress into main

### PR DESCRIPTION
**Pull request by DanPaseltiner**
# PULL REQUEST

## Summary
This PR merges the work that @m-goggins and I have been doing on the side to migrate the POC TEFCA Viewer from Python/HTMX to TypeScript/NextJs and implement the new UI designs. 

## Related Issue
Fixes #1730 

## Additional Information
Anything else the review team should know?

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

File: containers/tefca-viewer/api-documentation.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+**Content** : Error will be displayed to user
+
+### Or
+
+**Condition** : Authentication is invalid
+
+**Code** : `401 UNAUTHORIZED`
+
+**Content** : Error will be displayed to user
```
Comment (m-goggins) [inline containers/tefca-viewer/api-documentation.md:43]: This is for the eCR Viewer. I think we should update this as a separate ticket but I would delete the text for now so we know to go back and fix this.

**Review by m-goggins — APPROVED** [review]
Just left a note about updating the documentation in another PR but this looks good to me!

File: containers/tefca-viewer/api-documentation.md (hunk 1 above)
Comment (DanPaseltiner) [reply containers/tefca-viewer/api-documentation.md:43]: Thanks! I did create a ticket yesterday to update the README, but for forgot about the API docs. We should also just generate swagger docs like we do for all of the other Python based building blocks. I'll remove.

**Comment by m-goggins** [comment]
Not sure if you want to include it in this PR or another, but I found the cancer use case query that I wrote, [here](https://github.com/CDCgov/phdi/blob/tefca-ts-api-refactor/containers/tefca-viewer-nextjs/src/app/patient-search/patient_search.ts#L277-L331). I could add it in after this PR lands as a way to orient myself to the changes you've made.

**Comment by DanPaseltiner** [comment]
> Not sure if you want to include it in this PR or another, but I found the cancer use case query that I wrote, [here](https://github.com/CDCgov/phdi/blob/tefca-ts-api-refactor/containers/tefca-viewer-nextjs/src/app/patient-search/patient_search.ts#L277-L331). I could add it in after this PR lands as a way to orient myself to the changes you've made.

Let's do it in a separate PR #1720 is already stubbed for this. Thanks!

**Pull request merged**